File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        List<Integer> answer = new ArrayList<Integer>();
        
        for (int i : arr){
            if (i % divisor == 0){
                answer.add(i);
            }
        }
            
        Collections.sort(answer);
        
        if (answer.size() == 0) answer.add(-1);
            
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.kt
class Solution {
    fun solution(arr: IntArray, divisor: Int): IntArray {
        var answer = ArrayList<Int>()
        for(i in 0 until arr.size){
            if(arr[i]%divisor==0){
                answer.add(arr[i])
            }
        }
        return if(answer.size==0) intArrayOf(-1) else answer.toIntArray().sortedArray()
    }
}